extension/src/plots/model/collect.ts

```diff
@@ -93,6 +114,7 @@ const getCustomPlotData = (
   const paramPath = getFullValuePath(ColumnType.PARAMS, param)
 
   const values = getValues(experiments, metricPath, paramPath, renderLastIds)
+  const filteredColorScale = filterColorScale(completeColorScale, values)
```

Comment:
[Q] Is there a reason to do this after collecting `completeColorScale`? Would be good to have all of the transformation/collection in the same place.

Reply:
Originally, I was trying to avoid looping over all the experiments for each plot, but I now see we're still looping over `completeColorScale` anyway 😄. I'll update the code to create a single color scale per plot.